### ISSUE TITLE
repl_aae_fullsync test intermittent failures

### ISSUE THREAD

**Issue by cmeiklejohn** [basho/riak_repl#492]
Here's the details of one, which is failing when starting fullsync back up:

```
2013-12-17 20:32:20.924 [info] <0.25854.0>@riak_repl2_fscoordinator:determine_best_partition:616 starting partition search
2013-12-17 20:32:20.926 [info] <0.25854.0>@riak_repl2_fscoordinator:determine_best_partition:620 No partition in the queue that will not exceed a limit; will try again later.
2013-12-17 20:32:20.926 [info] <0.25854.0>@riak_repl2_fscoordinator:handle_socket_msg:512 Partition 867766597165223607683437869425293042920709947392 is too busy on cluster "B" at node 'dev6@127.0.0.1'
2013-12-17 20:32:20.926 [info] <0.25854.0>@riak_repl2_fscoordinator:determine_best_partition:616 starting partition search
2013-12-17 20:32:20.930 [info] <0.25854.0>@riak_repl2_fscoordinator:determine_best_partition:620 No partition in the queue that will not exceed a limit; will try again later.
2013-12-17 20:32:20.930 [info] <0.25854.0>@riak_repl2_fscoordinator:handle_socket_msg:512 Partition 433883298582611803841718934712646521460354973696 is too busy on cluster "B" at node 'dev6@127.0.0.1'
2013-12-17 20:32:20.930 [info] <0.25854.0>@riak_repl2_fscoordinator:determine_best_partition:616 starting partition search
2013-12-17 20:32:20.932 [info] <0.25854.0>@riak_repl2_fscoordinator:determine_best_partition:620 No partition in the queue that will not exceed a limit; will try again later.
2013-12-17 20:32:20.932 [info] <0.25854.0>@riak_repl2_fscoordinator:handle_socket_msg:512 Partition 137015778499772148581595453067151533092743675904 is too busy on cluster "B" at node 'dev6@127.0.0.1'
2013-12-17 20:32:20.932 [info] <0.25854.0>@riak_repl2_fscoordinator:determine_best_partition:616 starting partition search
2013-12-17 20:32:20.935 [info] <0.25854.0>@riak_repl2_fscoordinator:determine_best_partition:620 No partition in the queue that will not exceed a limit; will try again later.
2013-12-17 20:32:20.935 [info] <0.25854.0>@riak_repl2_fscoordinator:send_next_whereis_req:588 No partition available to start, no events outstanding, trying again later
```

@jcapricebasho @buddhisthead

**Comment by cmeiklejohn** [comment]
Looks like what @jcapricebasho was talking about, or at least similar to it.  Sink cluster is basically stalled with the same lock being held by a process that's doesn't appear to be doing anything:

```
Eshell V5.9.1  (abort with ^G)
(dev6@127.0.0.1)1> riak_repl2_fs_node_reserver:reserve(433883298582611803841718934712646521460354973696).
busy
(dev6@127.0.0.1)2> supervisor:which_children(riak_repl2_fssink_sup).
[{undefined,<0.6452.0>,worker,[riak_repl2_fssink]}]
(dev6@127.0.0.1)3> application:get_env(riak_repl, max_fssink_node).
undefined
(dev6@127.0.0.1)4> riak_repl2_fs_node_reserver:reserve(433883298582611803841718934712646521460354973696).
busy
(dev6@127.0.0.1)5> supervisor:which_children(riak_repl2_fssink_sup).
[{undefined,<0.6452.0>,worker,[riak_repl2_fssink]}]
(dev6@127.0.0.1)6> supervisor:which_children(riak_repl2_fssink_sup).
[{undefined,<0.6452.0>,worker,[riak_repl2_fssink]}]
(dev6@127.0.0.1)7> supervisor:which_children(riak_repl2_fssink_sup).
[{undefined,<0.6452.0>,worker,[riak_repl2_fssink]}]
(dev6@127.0.0.1)8> riak_repl2_fs_node_reserver:reserve(433883298582611803841718934712646521460354973696).
busy
(dev6@127.0.0.1)9>
```

**Comment by cmeiklejohn** [comment]
```
(dev6@127.0.0.1)17> process_info(Pid).
[{current_function,{gen_server,loop,6}},
 {initial_call,{proc_lib,init_p,5}},
 {status,waiting},
 {message_queue_len,0},
 {messages,[]},
 {links,[<0.2694.0>]},
 {dictionary,[{'$ancestors',[riak_repl2_fssink_sup,
                             riak_repl_sup,<0.665.0>]},
              {'$initial_call',{riak_repl2_fssink,init,1}}]},
 {trap_exit,false},
 {error_handler,error_handler},
 {priority,normal},
 {group_leader,<0.664.0>},
 {total_heap_size,233},
 {heap_size,233},
 {stack_size,9},
 {reductions,2947},
 {garbage_collection,[{min_bin_vheap_size,46368},
                      {min_heap_size,233},
                      {fullsweep_after,0},
                      {minor_gcs,0}]},
 {suspending,[]}]
```

```
(dev6@127.0.0.1)28> [process_info(P) || P <- proplists:get_value(links, process_info(Pid))].
[[{registered_name,riak_repl2_fssink_sup},
  {current_function,{gen_server,loop,6}},
  {initial_call,{proc_lib,init_p,5}},
  {status,waiting},
  {message_queue_len,0},
  {messages,[]},
  {links,[<0.2440.0>,<0.6452.0>]},
  {dictionary,[{'$ancestors',[riak_repl_sup,<0.665.0>]},
               {'$initial_call',{supervisor,riak_repl2_fssink_sup,1}}]},
  {trap_exit,true},
  {error_handler,error_handler},
  {priority,normal},
  {group_leader,<0.664.0>},
  {total_heap_size,233},
  {heap_size,233},
  {stack_size,9},
  {reductions,446762},
  {garbage_collection,[{min_bin_vheap_size,46368},
                       {min_heap_size,233},
                       {fullsweep_after,0},
                       {minor_gcs,0}]},
  {suspending,[]}]]
```

**Comment by cmeiklejohn** [comment]
It appears that 6452 (fssink) launches 6453, which is a aae_sink process, but when that sink is terminated the fssink never dies.

```
2013-12-17 20:11:28.586 [info] <0.5791.0>@riak_repl_aae_sink:process_msg:160 got complete
2013-12-17 20:11:28.586 [info] <0.5791.0>@riak_repl_aae_sink:terminate:112 Terminating.
2013-12-17 20:11:28.586 [info] <0.5790.0>@riak_repl2_fssink:handle_cast:130 Fullsync of partition complete.
2013-12-17 20:11:32.825 [info] <0.6450.0>@riak_core_service_mgr:try_ssl:367 "B" and "A" agreed to not use SSL
2013-12-17 20:11:32.825 [info] <0.6452.0>@riak_repl2_fssink:init:93 Negotiated {fullsync,{2,0},{2,0}} with ver w1
2013-12-17 20:11:32.825 [info] <0.6452.0>@riak_repl2_fssink:init:99 fullsync connection (ver w1) from cluster "A"
2013-12-17 20:11:32.826 [info] <0.6452.0>@riak_repl2_fssink:handle_info:164 Got caps: [{strategy,aae}]
2013-12-17 20:11:32.826 [info] <0.6452.0>@riak_repl2_fssink:handle_info:166 Common strategy: aae
2013-12-17 20:11:32.826 [info] <0.6453.0>@riak_repl_aae_sink:init:44 Starting AAE fullsync sink worker
2013-12-17 20:11:32.827 [info] <0.6453.0>@riak_repl_aae_sink:handle_info:90 AAE sink connection to "A" closed
2013-12-17 20:11:32.827 [info] <0.6453.0>@riak_repl_aae_sink:terminate:112 Terminating.
```

**Comment by cmeiklejohn** [comment]
Addressed by #493.
